doc/source/algorithms.rst

```diff
@@ -163,6 +163,7 @@ The following is an example that shows how to use this function:
 The result has several fields, among which the following are most useful:
 
 * ``parents[i]``:  the parent vertex of the i-th vertex. The parent of each source vertex is itself.
+* ``hasparent[i]``:  ``true`` if the i-th vertex has a parent, and ``false`` otherwise. When ``hasparent[i] == false``, it means that the vertex at index ``i`` isn't reachable from any source. Note that ``hasparent[i] == true`` for all source vertices.
```

Comment:
Yes. Technically, a node's parent is not itself unless there's an explicit edge between the node and itself, and that wouldn't ever exist (except possibly with A* and negative weights) in a shortest path calculation.

I don't know the original reason for making the source node its own parent, but it's not consistent with other packages (e.g., networkx) and will require an explicit check to ensure we're at the end of the (reversed) path so that we don't loop.

If we made this change, then the path loop can terminate on "hasparent=false".